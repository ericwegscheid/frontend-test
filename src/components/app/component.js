import React, { Component, Fragment } from 'react'
import { MainView } from '../main-view'

export class AppComponent extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return <MainView />
  }
}
