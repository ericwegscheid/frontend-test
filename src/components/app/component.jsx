import React, { Component } from 'react'
import { MainView } from '../../components/main-view'
import './styles'

export class AppComponent extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return <MainView />
  }
}
