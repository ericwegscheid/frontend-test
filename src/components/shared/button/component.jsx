import React from 'react'
import './styles'

export const Button = props => {
  return (
    <button {...props}>
      { props.children }
    </button>
  )
}
