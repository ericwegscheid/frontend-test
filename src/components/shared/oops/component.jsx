import React from 'react'
import './styles'

export const Oops = props => {
  return (
    <div className="oops" {...props}>
      <div className="space-invader"></div>
      <h1>oops</h1>
      <h3>failure to contact the mothership</h3>
    </div>
  )
}
