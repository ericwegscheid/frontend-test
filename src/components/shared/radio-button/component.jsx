import { isFunction } from 'lodash'
import React, { Component } from 'react'
import './styles'

export class RadioButton extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isActive: props.isActive,
      isDisabled: props.isDisabled,
    }
  }

  componentWillReceiveProps(nextProps) {
    const shouldUpdate =
      this.state.isActive !== nextProps.isActive ||
      this.state.isDisabled !== nextProps.isDisabled


    if (shouldUpdate) {
      this.setState({
        isActive: nextProps.isActive,
        isDisabled: nextProps.isDisabled,
      })
    }

    return shouldUpdate
  }

  onClick() {
    this.setState(state => ({
      isActive: !state.isActive,
    }))

    if (isFunction(this.props.onClick)) {
      this.props.onClick()
    }
  }

  render() {
    const { isActive, isDisabled } = this.state

    return <div
      className={`control ${isDisabled ? 'disabled' : ''}`}
      onClick={this.onClick.bind(this)}
    >
      <div
        className={`radio-button ${isActive ? 'active' : ''}`}
      ></div>
      <label>{this.props.label}</label>
    </div>
  }
}
