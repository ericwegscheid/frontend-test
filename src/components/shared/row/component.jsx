import React from 'react'
import './styles'

export const Row = props => {
  return (
    <div className="row" {...props}>
      <div className="row-content">
        { props.children }
      </div>
    </div>
  )
}
