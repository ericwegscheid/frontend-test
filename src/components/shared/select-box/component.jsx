import { isFunction, map } from 'lodash'
import React, { Component } from 'react'
import './styles'

export class SelectBox extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isOpen: false,
      options: {
        ...props.options,
      },
      selected: props.selected,
    }
  }

  componentWillReceiveProps(nextProps) {
    const shouldUpdate = this.state.selected !== nextProps.selected

    if (shouldUpdate) {
      this.setState({
        selected: nextProps.selected,
      })
    }

    return shouldUpdate
  }

  onClick() {
    this.setState(state => ({
      isOpen: !state.isOpen,
    }))
  }

  onSelectItem(selected) {
    this.setState(state => ({
      selected: selected,
    }))

    if (isFunction(this.props.onSelectItem)) {
      this.props.onSelectItem(selected)
    }
  }

  render() {
    const { isOpen, selected } = this.state

    return <div
      className={`control select-box ${isOpen ? 'open' : ''}`}
      onClick={this.onClick.bind(this)}
    >
      <ul>
        {
          map(this.props.options, v =>
            <li
              key={v.key}
              className={selected.key === v.key ? 'selected' : ''}
              onClick={this.onSelectItem.bind(this, v)}
            >
              <span>{v.value}</span>
            </li>
          )
        }
      </ul>
      <label>{this.props.label}<span>{selected.value}</span></label>
    </div>
  }
}
