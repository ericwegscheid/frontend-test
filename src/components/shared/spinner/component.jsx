import React from 'react'
import './styles'

export const Spinner = () => {
  return (
    <div className="spinner">
      <div className="mask"></div>
    </div>
  )
}
