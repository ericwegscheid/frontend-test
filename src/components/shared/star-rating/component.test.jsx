import React from 'react'
import { StarRating } from './component.jsx'
import { shallow } from 'enzyme'

describe('StarRating component', () => {
  let component

  const renderComponent = props => {
    component = shallow(props)
  }

  it('should renderer snapshot', () => {
    renderComponent(
      <StarRating />
    )

    expect(component.html()).toMatchSnapshot()
  })

  it('should renderer 3 full stars and 2 empty stars', () => {
    renderComponent(
      <StarRating rating={3}/>
    )

    expect(component.find('Star[fill="full"]').length).toBe(3)
    expect(component.find('Star[fill="half"]').length).toBe(0)
    expect(component.find('Star[fill="empty"]').length).toBe(2)

    expect(component.html()).toMatchSnapshot()
  })

  it('should renderer 2 full stars, 1 half star, and 2 empty stars', () => {
    renderComponent(
      <StarRating rating={2.3}/>
    )

    expect(component.find('Star[fill="full"]').length).toBe(2)
    expect(component.find('Star[fill="half"]').length).toBe(1)
    expect(component.find('Star[fill="empty"]').length).toBe(2)

    expect(component.html()).toMatchSnapshot()
  })

  it('should renderer 5 full stars', () => {
    renderComponent(
      <StarRating rating={6}/>
    )

    expect(component.find('Star[fill="full"]').length).toBe(5)
    expect(component.find('Star[fill="half"]').length).toBe(0)
    expect(component.find('Star[fill="empty"]').length).toBe(0)

    expect(component.html()).toMatchSnapshot()
  })

  it('should renderer 5 empty stars', () => {
    renderComponent(
      <StarRating rating={-2}/>
    )

    expect(component.find('Star[fill="full"]').length).toBe(0)
    expect(component.find('Star[fill="half"]').length).toBe(0)
    expect(component.find('Star[fill="empty"]').length).toBe(5)

    expect(component.html()).toMatchSnapshot()
  })
})
