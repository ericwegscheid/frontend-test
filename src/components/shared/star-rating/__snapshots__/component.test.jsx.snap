// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating component should renderer 2 full stars, 1 half star, and 2 empty stars 1`] = `"<div class=\\"star-rating\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.2168,0 8.58203,8.167969 0,8.148439 l 6.95508,5.029297 -2.66992,8.15625 6.93164,-5.060547 6.93164,5.060547 -2.66992,-8.15625 6.95312,-5.029297 -8.58203,0.01953 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.2168,0 8.58203,8.167969 0,8.148439 l 6.95508,5.029297 -2.66992,8.15625 6.93164,-5.060547 6.93164,5.060547 -2.66992,-8.15625 6.95312,-5.029297 -8.58203,0.01953 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.21529,0 8.58134,8.167124 0,8.148294 6.95342,13.177188 4.28386,21.3326 11.21529,16.273474 18.14672,21.3326 15.477,13.177188 22.43042,8.148294 13.84924,8.167124 Z m 0,3.696328 1.80836,5.607093 5.8912,-0.01282 -4.77374,3.452418 1.83289,5.599179 -4.75871,-3.473312 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.21484,0 8.58089,8.166944 0,8.148344 l 6.95306,5.028634 -2.66909,8.155056 6.93087,-5.058606 6.93134,5.058606 -2.66909,-8.155056 6.95307,-5.028634 -8.58086,0.0186 z m 0,3.696417 1.80867,5.606375 5.8911,-0.0124 -4.77385,3.452501 1.83296,5.599141 -4.75888,-3.473172 -4.75837,3.473172 1.83295,-5.599141 -4.77387,-3.452501 5.89113,0.0124 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.21484,0 8.58089,8.166944 0,8.148344 l 6.95306,5.028634 -2.66909,8.155056 6.93087,-5.058606 6.93134,5.058606 -2.66909,-8.155056 6.95307,-5.028634 -8.58086,0.0186 z m 0,3.696417 1.80867,5.606375 5.8911,-0.0124 -4.77385,3.452501 1.83296,5.599141 -4.75888,-3.473172 -4.75837,3.473172 1.83295,-5.599141 -4.77387,-3.452501 5.89113,0.0124 z\\"></path></svg></div>"`;

exports[`StarRating component should renderer 3 full stars and 2 empty stars 1`] = `"<div class=\\"star-rating\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.2168,0 8.58203,8.167969 0,8.148439 l 6.95508,5.029297 -2.66992,8.15625 6.93164,-5.060547 6.93164,5.060547 -2.66992,-8.15625 6.95312,-5.029297 -8.58203,0.01953 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.2168,0 8.58203,8.167969 0,8.148439 l 6.95508,5.029297 -2.66992,8.15625 6.93164,-5.060547 6.93164,5.060547 -2.66992,-8.15625 6.95312,-5.029297 -8.58203,0.01953 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.2168,0 8.58203,8.167969 0,8.148439 l 6.95508,5.029297 -2.66992,8.15625 6.93164,-5.060547 6.93164,5.060547 -2.66992,-8.15625 6.95312,-5.029297 -8.58203,0.01953 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.21484,0 8.58089,8.166944 0,8.148344 l 6.95306,5.028634 -2.66909,8.155056 6.93087,-5.058606 6.93134,5.058606 -2.66909,-8.155056 6.95307,-5.028634 -8.58086,0.0186 z m 0,3.696417 1.80867,5.606375 5.8911,-0.0124 -4.77385,3.452501 1.83296,5.599141 -4.75888,-3.473172 -4.75837,3.473172 1.83295,-5.599141 -4.77387,-3.452501 5.89113,0.0124 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.21484,0 8.58089,8.166944 0,8.148344 l 6.95306,5.028634 -2.66909,8.155056 6.93087,-5.058606 6.93134,5.058606 -2.66909,-8.155056 6.95307,-5.028634 -8.58086,0.0186 z m 0,3.696417 1.80867,5.606375 5.8911,-0.0124 -4.77385,3.452501 1.83296,5.599141 -4.75888,-3.473172 -4.75837,3.473172 1.83295,-5.599141 -4.77387,-3.452501 5.89113,0.0124 z\\"></path></svg></div>"`;

exports[`StarRating component should renderer 5 empty stars 1`] = `"<div class=\\"star-rating\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.21484,0 8.58089,8.166944 0,8.148344 l 6.95306,5.028634 -2.66909,8.155056 6.93087,-5.058606 6.93134,5.058606 -2.66909,-8.155056 6.95307,-5.028634 -8.58086,0.0186 z m 0,3.696417 1.80867,5.606375 5.8911,-0.0124 -4.77385,3.452501 1.83296,5.599141 -4.75888,-3.473172 -4.75837,3.473172 1.83295,-5.599141 -4.77387,-3.452501 5.89113,0.0124 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.21484,0 8.58089,8.166944 0,8.148344 l 6.95306,5.028634 -2.66909,8.155056 6.93087,-5.058606 6.93134,5.058606 -2.66909,-8.155056 6.95307,-5.028634 -8.58086,0.0186 z m 0,3.696417 1.80867,5.606375 5.8911,-0.0124 -4.77385,3.452501 1.83296,5.599141 -4.75888,-3.473172 -4.75837,3.473172 1.83295,-5.599141 -4.77387,-3.452501 5.89113,0.0124 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.21484,0 8.58089,8.166944 0,8.148344 l 6.95306,5.028634 -2.66909,8.155056 6.93087,-5.058606 6.93134,5.058606 -2.66909,-8.155056 6.95307,-5.028634 -8.58086,0.0186 z m 0,3.696417 1.80867,5.606375 5.8911,-0.0124 -4.77385,3.452501 1.83296,5.599141 -4.75888,-3.473172 -4.75837,3.473172 1.83295,-5.599141 -4.77387,-3.452501 5.89113,0.0124 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.21484,0 8.58089,8.166944 0,8.148344 l 6.95306,5.028634 -2.66909,8.155056 6.93087,-5.058606 6.93134,5.058606 -2.66909,-8.155056 6.95307,-5.028634 -8.58086,0.0186 z m 0,3.696417 1.80867,5.606375 5.8911,-0.0124 -4.77385,3.452501 1.83296,5.599141 -4.75888,-3.473172 -4.75837,3.473172 1.83295,-5.599141 -4.77387,-3.452501 5.89113,0.0124 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.21484,0 8.58089,8.166944 0,8.148344 l 6.95306,5.028634 -2.66909,8.155056 6.93087,-5.058606 6.93134,5.058606 -2.66909,-8.155056 6.95307,-5.028634 -8.58086,0.0186 z m 0,3.696417 1.80867,5.606375 5.8911,-0.0124 -4.77385,3.452501 1.83296,5.599141 -4.75888,-3.473172 -4.75837,3.473172 1.83295,-5.599141 -4.77387,-3.452501 5.89113,0.0124 z\\"></path></svg></div>"`;

exports[`StarRating component should renderer 5 full stars 1`] = `"<div class=\\"star-rating\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.2168,0 8.58203,8.167969 0,8.148439 l 6.95508,5.029297 -2.66992,8.15625 6.93164,-5.060547 6.93164,5.060547 -2.66992,-8.15625 6.95312,-5.029297 -8.58203,0.01953 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.2168,0 8.58203,8.167969 0,8.148439 l 6.95508,5.029297 -2.66992,8.15625 6.93164,-5.060547 6.93164,5.060547 -2.66992,-8.15625 6.95312,-5.029297 -8.58203,0.01953 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.2168,0 8.58203,8.167969 0,8.148439 l 6.95508,5.029297 -2.66992,8.15625 6.93164,-5.060547 6.93164,5.060547 -2.66992,-8.15625 6.95312,-5.029297 -8.58203,0.01953 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.2168,0 8.58203,8.167969 0,8.148439 l 6.95508,5.029297 -2.66992,8.15625 6.93164,-5.060547 6.93164,5.060547 -2.66992,-8.15625 6.95312,-5.029297 -8.58203,0.01953 z\\"></path></svg><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"84.781006\\" height=\\"80.632385\\" viewBox=\\"0 0 22.431642 21.333984\\" version=\\"1.1\\"><path d=\\"M 11.2168,0 8.58203,8.167969 0,8.148439 l 6.95508,5.029297 -2.66992,8.15625 6.93164,-5.060547 6.93164,5.060547 -2.66992,-8.15625 6.95312,-5.029297 -8.58203,0.01953 z\\"></path></svg></div>"`;

exports[`StarRating component should renderer snapshot 1`] = `"<div class=\\"star-rating\\"></div>"`;
